wit/network.wit

```diff
@@ -13,11 +13,77 @@ default interface network {
 	drop-network: func(this: network)
 
 
-
+	/// Error codes.
+	/// 
+	/// In theory, every API can return any error code.
+	/// In practice, API's typically only return the errors documented per API
+	/// combined with a couple of errors that are always possible:
+	/// - `unknown`
+	/// - `access-denied`
+	/// - `not-supported`
+	/// - `resource-limit-reached`
+	/// - `out-of-memory`
+	/// 
+	/// See each individual API for what the POSIX equivalents are. They sometimes differ per API.
 	enum error {
+		// ### GENERAL ERRORS ###
+
+		/// Unknown error
 		unknown,
-		again,
-		// TODO ...
+
+		/// POSIX equivalent: EACCES, EPERM
+		access-denied,
+
+		/// POSIX equivalent: EOPNOTSUPP
+		not-supported,
+
+		/// POSIX equivalent: EMFILE, ENFILE, ENOMEM, ENOBUFS, EAI_MEMORY
+		resource-limit-reached,
```

Comment:
Why is `ENOMEM` listed here, when it seems to correspond to `out-of-memory` too?

Reply:
The idea is that `resource-limit-reached` can be used by WASI hosts as a catch-all for any kind of resource limit. (not just file descriptors and memory). Including limits completely foreign to UNIX, like: "A wasm plugin may only make 2000 external connections per month"

`out-of-memory` is interpreted as an optional specialization of `resource-limit-reached` and left up to hosts whether or not they want to communicate _which_ limit has been hit. In other words, `resource-limit-reached` might also be returned for out of memory.

Reply:
It might be less confusing if I were to just remove the POSIX equivalents from the `resource-limit-reached` documentation.

Reply:
See d28139697d720ca8e2277cbac4ef77285afc20ea

Reply:
This specialization relationship isn't obvious; could you add the content of your comment above to the documentation?

Reply:
I've given this some thoughts over the last couple of days. TBH, I'm not quite convinced of my own idea.

`resource-limit-reached` feels like a solution to a problem that doesn't exist (yet, or never will).
Given that there is no equivalent of it in POSIX, it would be awkward for libc to translate back into POSIX error codes. Most likely, it would trip up existing applications. Additionally, the subtyping relationship is completely foreign to anyone familiar with the native socket APIs.

In my [latest commit](https://github.com/WebAssembly/wasi-sockets/pull/18/commits/b9a85577328a8023102b8ba9003cd323c4171fd5) I've backtracked on myself and replaced it with `new-socket-limit` (_"A new socket resource could not be created because of a system limit."_)

Let me know if your OK with this, so I can merge this.

Reply:
Merging this so we can move on.